### DISCUSSION TITLE
Is it possible to parse links in the message field?

### DISCUSSION THREAD

**Discussion by mietcls** [ugent-library/biblio-backend#1324] [Q&A]
# Enhancement
Is it possible to parse links in the message field?

We can't integrate a full-blown messaging system at this point, and it might still take a while before we can do so. However, helping people complete their withdrawn publications and datasets is not very user-friendly as the links provided by the helpdesk are not clickable.

Is this low-hanging fruit that can improve the situation temporarily without too much maintenance?

## Case 1: HTTPS
![Screenshot 2023-11-07 at 15 29 27](https://github.com/ugent-library/biblio-backoffice/assets/1572885/b1972cc5-23b2-45cc-8da8-61ce966b193e)

## Case 2: bit.ly
![Screenshot 2023-11-07 at 15 29 46](https://github.com/ugent-library/biblio-backoffice/assets/1572885/473b6650-fa15-4a1e-896c-3128df204534)

**Comment by nicolasfranck** [comment]
some "markdown" interpretation, like github does? Of course that will have some unexpected results too (that `**` being interpreted as "bold")

**Comment by mietcls** [comment]
That would actually be nice. These are (mostly) temporary messages in the back office that would not be published to the outside world / the front office.

**Comment by mietcls** [comment]
However; it's only nice if it can be built with durability in mind within a reasonable amount of time, without a lot of maintenance / bugs afterwards.

**Comment by nics** [comment]
braindead quick and dirty solution: strip tags for security reasons; loop over all words in the message; add link if `url.Parse` succeeds; render message as html

**Reply by nicolasfranck** [reply]
😆
```
package main

import (
	"fmt"
	"net/url"
)

func main() {
	u, err := url.Parse("1")
	fmt.Printf("url: %+v, err: %+v\n", u, err)
}
```

returns

```
url: 1, err: <nil>
```

**Reply by nicolasfranck** [reply]
Better with `url.ParseRequestURI`. Apparently we use `url.Parse` a lot ..

**Comment by netsensei** [comment]
This is a safe HTML sanitizer: https://github.com/microcosm-cc/bluemonday

This library is policy based. Comes with a few pre-defined policies like `strict` (absolutely no tags whatsoever) and a sensible default (whitelist sensible tags like bold or italic). You can also define your own policy / whitelist of allowed elements.

It does URL parsing intelligently via `p.RequireParseableURLs(true)`
